onnxruntime/contrib_ops/cpu/bert/group_query_attention_helper.h

```diff
@@ -11,18 +11,19 @@ namespace onnxruntime {
 namespace contrib {
 namespace group_query_attention_helper {
 
-Status CheckInputs(const Tensor* query,
-                   const Tensor* key,
-                   const Tensor* value,
-                   const Tensor* past_key,
-                   const Tensor* past_value,
-                   const Tensor* cos_cache,
-                   const Tensor* sin_cache,
+template <typename T = Tensor>
```

Comment:
template seems not necessary

a more straightforward way is to keep delaration only in .h, and put the implementation in .cc

Reply:
That is true. However doing so requires modifying the code, moving definition in a new source file and keep the declaration in the header file. It is not a generally good practice to put code like this in the header file. I am trying to work with minimal changes. multihead_attention_helper.h works the same way.